tardis/utilities/utils.py

```diff
@@ -66,6 +67,32 @@ def csv_parser(
             }
 
 
+KeyType = TypeVar("KeyType")
+
+
+def deep_update(
+    original_mapping: Mapping[KeyType, Any],
+    mapping_update: Mapping[KeyType, Any],
+) -> Mapping[KeyType, Any]:
+    from collections.abc import Mapping
+
+    updated_mapping = deepcopy(original_mapping)
+    for key, value in mapping_update.items():
+        if isinstance(value, Mapping):
+            updated_mapping[key] = deep_update(value, original_mapping.get(key, {}))
+        elif isinstance(value, List):
```

Comment:
This seems to be fine to always check for the concrete class.
```suggestion
        elif isinstance(value, list):
```